pkg/services/annotations/annotationsimpl/composite_store.go

```diff
@@ -65,3 +79,14 @@ func (c *CompositeStore) GetTags(ctx context.Context, query *annotations.TagsQue
 
 	return annotations.FindTagsResult{Tags: res}, nil
 }
+
+// panicHandler is a helper function that recovers from a panic and logs the error,
+// and sets the error pointer to a new error if it is not nil.
+func panicHandler(logger log.Logger, storeType string, pErr *error) {
```

Comment:
```suggestion
func panicToError(logger log.Logger, storeType string, pErr *error) {
```
just a naming nit.

at first look, i didn't catch that we were converting the panic back to an error - i missed the named return parameter in `ForEachJob`. This makes things more readily apparent at a glance.